receiver/sqlserverreceiver/factory.go

```diff
@@ -64,9 +66,26 @@ func directDBConnectionEnabled(config *Config) bool {
 		string(config.Password) != ""
 }
 
+// Gets the end-result host and port by splitting `config.Server`, or keeping both `config.Server` and `config.Port`
+func getServerPort(config *Config) (host, port string) {
```

Comment:
Do you need to return host? Maybe rename the function if you do, as it does more than getting the port.